### PR TITLE
Added the vote command

### PR DESCRIPTION
**Pull request by Double-Felix**
Supports default and custom emojis. Also added comments to help clarify stuff in the future

### CODE REVIEW

File: commanderbot_ext/vote/__init__.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+from discord.ext import commands
+from commanderbot_ext.vote.vote_cog import VoteCog
+
+
+def setup(bot: commands.Bot):
+    bot.add_cog(VoteCog(bot))
```
Comment (Arcensoth) [inline commanderbot_ext/vote/__init__.py:6]: Make sure you run the auto-formatter before committing. There should be a newline at the end of every file.

File: commanderbot_ext/vote/vote_cog.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        for emoji in self.get_emojis(ctx.message, ctx) or ("👍", "👎")[:20]:
+            try:
+                # Attempt to react with the emoji specified
+                await ctx.message.add_reaction(emoji)
+
+            # If any error occurred, then don't bother adding the reaction
+            except Exception as error:
+                self._log.info(
+                    f"Couldn't add reaction {emoji} to message {ctx.message.id}\nError is as follows:")
+                self._log.info(error)
```
Comment (Arcensoth) [inline commanderbot_ext/vote/vote_cog.py:55]: ~~Same here; the auto-formatter should have placed a newline at the end of the file.~~ I think GitHub just doesn't show the empty newline here, but just as a reminder make sure you run the import sorter and auto-formatter after you make the other changes.

File: commanderbot_ext/vote/vote_cog.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+        # Get all emojis in the message and return a generator.
+        # All default emojis become unicode which can be reacted with no problem, and custom emojis become <:name:ID> which can be used normally
+        emoji_matches = emoji_patterns.findall(message.clean_content)
+        yield from emoji_matches
+
+    @command(name="vote")
+    async def cmd_vote(self, ctx: discord.Message):
+        # If no emojis were specified, then use the default ones, then iterate over them (reaction cap is 20)
+        for emoji in self.get_emojis(ctx.message, ctx) or ("👍", "👎")[:20]:
```
Comment (Arcensoth) [inline commanderbot_ext/vote/vote_cog.py:46]: Does this not take up to 20 elements from just `("👍", "👎")`? I believe you need brackets here.

**Review by Arcensoth — CHANGES_REQUESTED** [review]
Also remember to add to CHANGELOG.md as discussed.

File: commanderbot_ext/vote/vote_cog.py (hunk 3 above)
Comment (Double-Felix) [reply commanderbot_ext/vote/vote_cog.py]: Fixed in latest commit

File: commanderbot_ext/vote/vote_cog.py (hunk 2 above)
Comment (Double-Felix) [reply commanderbot_ext/vote/vote_cog.py:54]: Fixed in latest commit

File: commanderbot_ext/vote/__init__.py (hunk 1 above)
Comment (Double-Felix) [reply commanderbot_ext/vote/__init__.py:6]: Fixed in latest commit

**Review by Arcensoth — CHANGES_REQUESTED** [review]

File: commanderbot_ext/vote/vote_cog.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        for emoji in (self.get_emojis(ctx.message, ctx) or ("👍", "👎"))[:20]:
+            try:
+                # Attempt to react with the emoji specified
+                await ctx.message.add_reaction(emoji)
+
+            # If any error occurred, then don't bother adding the reaction
+            except Exception as error:
+                self._log.info(
+                    f"Couldn't add reaction {emoji} to message {ctx.message.id}\nError is as follows:")
+                self._log.info(error)
```
[the remark this answers is not in the document]
Comment (Double-Felix) [reply commanderbot_ext/vote/vote_cog.py]: Fixed in latest commit

**Review by Arcensoth — APPROVED** [review]

**Pull request merged**